src/cephadm/cephadm

```diff
@@ -5212,6 +5257,462 @@ def command_gather_facts():
+                except json.decoder.JSONDecodeError:
+                    errors.append("host-facts received invalid JSON")
+                    logger.warning(errors[-1])
+                    data = {}
+                with self.cephadm_cache_lock:
+                    self.cephadm_cache['host'] = {
+                        "scrape_timestamp": s_time,
+                        "scrape_duration_secs": elapsed,
+                        "scrape_errors": errors,
+                        "data": data,                    
+                    }
+                logger.debug(f"completed host-facts scrape - {elapsed}s")
+            
+            time.sleep(CephadmDaemon.loop_delay)
+            ctr += CephadmDaemon.loop_delay
+        logger.info("host-facts thread stopped")
+
+    def _scrape_ceph_volume(self, refresh_interval=15):
+        args.command = "inventory --format=json".split()
+        args.fsid = self.fsid
+        args.log_output = False
+        ctr = 0
+        while True:
+            if self.stop:
+                break
+            if ctr >= refresh_interval:
+                ctr = 0
+                logger.debug("executing ceph-volume scrape")
+                errors = []
+                s_time = time.time()
+                stream = io.StringIO()
+
+                with redirect_stdout(stream):
+                    command_ceph_volume()
+                elapsed = time.time() - s_time
+
+                # if the call to ceph-volume returns junk with the 
+                # json, it won't parse
+                stdout = stream.getvalue()
+
+                if stdout:
+                    try:
+                        data = json.loads(stdout)
+                    except json.decoder.JSONDecodeError:
+                        errors.append("ceph-volume thread received bad json data")
+                        logger.warning(errors[-1])
+                        data = []
+                else:
+                    errors.append("ceph-volume didn't return any data")
+                    logger.warning(errors[-1])
+
+                with self.cephadm_cache_lock:
+                    self.cephadm_cache['disks'] = {
+                        "scrape_timestamp": s_time,
+                        "scrape_duration_secs": elapsed,
+                        "scrape_errors": errors,
+                        "data": data, 
+                    }
+                
+                logger.debug(f"completed ceph-volume scrape - {elapsed}s")
+            time.sleep(CephadmDaemon.loop_delay)
+            ctr += CephadmDaemon.loop_delay
+        
+        logger.info("ceph-volume thread stopped")
+
+    def _scrape_list_daemons(self, refresh_interval=20):
+        ctr = 0
+        while True:
+            if self.stop:
+                break
+            
+            if ctr >= refresh_interval:
+                ctr = 0
+                logger.debug("executing list-daemons scrape")
+                errors = []
```

Comment:
The 'errors' list is never updated here. Maybe it is worth to remove 'errors' , and use an empty list when needed.

I think it makes code more clear because if i see a variable i think that is going to be used for something.

Reply:
fair point - I'm missing a json test in that thread - which would populate the errors list (like the other threads). Will add.